From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Yu <davyu@nvidia.com>
Date: Fri, 8 Oct 2021 02:29:25 +0000
Subject: [PATCH 038/243] NVIDIA: SAUCE: net: phy: fix data corruption

Data corruption is happening in aqr107_config_init().
It's accessing address offset across struct phy_device boundary.
This is caused by misuse of SUPPORTED_* to linkmode_*_bit() api.
Change SUPPORTED_* to ETHTOOL_LINK_MODE_*_BIT for correct usage.

Bug 200781153
Bug 4088909
Bug 4090066

Change-Id: I6e2a50b330c34620c86f42aceddad046fcd3a658
Signed-off-by: David Yu <davyu@nvidia.com>
Signed-off-by: Revanth Kumar Uppala <ruppala@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2906776
Reviewed-by: Abhilash G <abhilashg@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>
Tested-by: Abhilash G <abhilashg@nvidia.com>

diff --git a/drivers/net/phy/aquantia/aquantia_main.c b/drivers/net/phy/aquantia/aquantia_main.c
index d492d5b99f9a..922fd352d2c3 100644
--- a/drivers/net/phy/aquantia/aquantia_main.c
+++ b/drivers/net/phy/aquantia/aquantia_main.c
@@ -934,8 +934,8 @@ static int aqr_gen1_config_init(struct phy_device *phydev)
 	}
 
 	/* Advertize flow control */
-	linkmode_set_bit(SUPPORTED_Pause, phydev->supported);
-	linkmode_set_bit(SUPPORTED_Asym_Pause, phydev->supported);
+	linkmode_set_bit(ETHTOOL_LINK_MODE_Pause_BIT, phydev->supported);
+	linkmode_set_bit(ETHTOOL_LINK_MODE_Asym_Pause_BIT, phydev->supported);
 	linkmode_copy(phydev->advertising, phydev->supported);
 
 	/* Configure flow control */
diff --git a/drivers/net/phy/realtek/realtek_main.c b/drivers/net/phy/realtek/realtek_main.c
index d011d10cfaf4..dddda6f5730d 100644
--- a/drivers/net/phy/realtek/realtek_main.c
+++ b/drivers/net/phy/realtek/realtek_main.c
@@ -661,8 +661,8 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 	}
 
 	/* Advertise Flow Control */
-	linkmode_set_bit(SUPPORTED_Pause, phydev->supported);
-	linkmode_set_bit(SUPPORTED_Asym_Pause, phydev->supported);
+	linkmode_set_bit(ETHTOOL_LINK_MODE_Pause_BIT, phydev->supported);
+	linkmode_set_bit(ETHTOOL_LINK_MODE_Asym_Pause_BIT, phydev->supported);
 	linkmode_copy(phydev->advertising, phydev->supported);
 
 	return genphy_soft_reset(phydev);
-- 
2.43.0

